src/core/tests/main.cpp

```diff
@@ -71,7 +71,9 @@ GTEST_API_ int main(int argc, char **argv)
 
     if (g_test_ret != 0)
     {
+#ifndef ENABLE_GCOV
```

Comment:
yes, it may crash if not use dsn_exit, but if use dsn_exit, the process will not generate '.gcda' file which contains the code coverage stats. So we use macro here to not use dsn_exit when for code coverage.